Makefile

```diff
@@ -0,0 +1,147 @@
+# ====================================================================
+# Copyright oVirt Authors
+# SPDX-License-Identifier: Apache-2.0
+# ====================================================================
+
+#
+# CUSTOMIZATION-BEGIN
+#
+# Keycloak version specification
+KEYCLOAK_VERSION="15.0.2"
+
+# RPM version specification
+RPM_VERSION="${KEYCLOAK_VERSION}"
+RPM_RELEASE="1"
+
+EXTRA_BUILD_FLAGS=
+BUILD_VALIDATION=1
+
+PACKAGE_NAME=ovirt-engine-keycloak
+
+PYTHON=$(shell which python3 2> /dev/null)
+PREFIX=/usr/local
+SYSCONF_DIR=$(PREFIX)/etc
+DATAROOT_DIR=$(PREFIX)/share
+PKG_DATA_DIR=$(DATAROOT_DIR)/ovirt-engine-keycloak
+KEYCLOAK_OVERLAY_ZIP="keycloak-overlay-$(KEYCLOAK_VERSION).zip"
+KEYCLOAK_OVERLAY_URL="https://github.com/keycloak/keycloak/releases/download/${KEYCLOAK_VERSION}/${KEYCLOAK_OVERLAY_ZIP}"
+#
+# CUSTOMIZATION-END
+#
+BUILD_FLAGS:=$(BUILD_FLAGS) $(EXTRA_BUILD_FLAGS)
+
+TARBALL=$(PACKAGE_NAME)-$(RPM_VERSION).tar.gz
+BUILD_FILE=tmp.built
+
+
+.SUFFIXES:
+.SUFFIXES: .in
+
+.in:
+	sed \
+	-e "s|@KEYCLOAK_VERSION@|$(KEYCLOAK_VERSION)|g" \
+	-e "s|@KEYCLOAK_OVERLAY_ZIP@|$(KEYCLOAK_OVERLAY_ZIP)|g" \
+	-e "s|@DATAROOT_DIR@|$(DATAROOT_DIR)|g" \
+	-e "s|@PKG_DATA_DIR@|$(PKG_DATA_DIR)|g" \
+	-e "s|@RPM_VERSION@|$(RPM_VERSION)|g" \
+	-e "s|@RPM_RELEASE@|$(RPM_RELEASE)|g" \
+	-e "s|@PACKAGE_NAME@|$(PACKAGE_NAME)|g" \
+	$< > $@
+
+
+GENERATED = \
+	build/python-check.sh \
+	ovirt-engine-keycloak.spec \
+	packaging/setup/ovirt_engine_setup/keycloak/config.py \
+	$(NULL)
+
+
+all:	\
+	generated-files \
+	validations \
+	$(BUILD_FILE) \
+	$(NULL)
+
+generated-files:	$(GENERATED)
+	chmod a+x build/python-check.sh
+
+$(BUILD_FILE):
+	touch $(BUILD_FILE)
+
+clean:
+	rm -rf $(BUILD_FILE)
+	rm -rf tmp.dev.flist
+	rm -rf $(GENERATED)
+	rm -f "$(PACKAGE_NAME)-*.tar.gz"
+
+install: \
+	all \
+	install-packaging-files \
+	$(NULL)
+
+.PHONY: ovirt-engine-keycloak.spec.in
+
+
+dist:	ovirt-engine-keycloak.spec \
+	download-keycloak \
+	$(NULL)
+
+	git ls-files | tar --files-from /proc/self/fd/0 -czf \
+		"$(TARBALL)" \
+		ovirt-engine-keycloak.spec \
+		$(KEYCLOAK_OVERLAY_ZIP)
+	@echo
+	@echo For distro specific packaging refer to http://www.ovirt.org/Build_Binary_Package
+	@echo
+
+download-keycloak:
+	if [ ! -f "$(KEYCLOAK_OVERLAY_ZIP)" ]; then \
```

Comment:
Perhaps make this conditional? Not sure what your plans are for RHV, but build processes there do not download their stuff from the net... Can be done later.

Reply:
You mean a switch to download it or to copy from some local filesystem location, right? (I will create an Issue to remember/consider it for future)
For RHV we do not expect Keycloak to be used at all. RH SSO integration will be expected to be configured by administrator if needed.